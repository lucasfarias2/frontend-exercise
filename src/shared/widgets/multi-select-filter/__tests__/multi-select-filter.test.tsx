import React from 'react';
import { render, fireEvent, waitForElement, cleanup } from '@testing-library/react';
import MultiSelectFilter from '../multi-select-filter';

afterEach(cleanup);

describe('<MultiSelectFilter />', () => {
  const selectedItems: TSelectedItems = ['Fantasy Selected'];
  const getSelectedItems = jest.fn();
  const removeItem = jest.fn();
  const selectItem = jest.fn();
  const setItemsFilter = jest.fn();
  const clearSelectedItems = jest.fn();

  const props = {
    data: ['Literatuur &amp; Romans', 'Thrillers', 'Fantasy', 'Kinderboeken', 'Young Adult'],
    filter: 'fantasy',
    selectedItems,
    getSelectedItems,
    removeItem,
    selectItem,
    setItemsFilter,
    clearSelectedItems,
  };

  it('renders correctly', () => {
    const { container } = render(<MultiSelectFilter {...props} />);
    expect(container).toMatchSnapshot();
    expect(getSelectedItems).toHaveBeenCalled();
  });

  it('calls setItemsFilter', () => {
    const { getByTestId } = render(<MultiSelectFilter {...props} />);
    fireEvent.click(getByTestId('button_apply'));
    expect(setItemsFilter).toHaveBeenCalled();
  });

  it('calls selectItem', () => {
    const { getByTestId } = render(<MultiSelectFilter {...props} />);
    fireEvent.click(getByTestId('Fantasy'));
    expect(selectItem).toHaveBeenCalled();
  });

  it('calls removeItem', () => {
    const { getByTestId } = render(<MultiSelectFilter {...props} />);
    fireEvent.click(getByTestId('Fantasy Selected'));
    expect(removeItem).toHaveBeenCalled();
  });

  it('calls clearSelectedITems', () => {
    const { getByTestId } = render(<MultiSelectFilter {...props} />);
    fireEvent.click(getByTestId('button_clear'));
    expect(clearSelectedItems).toHaveBeenCalled();
  });
});
