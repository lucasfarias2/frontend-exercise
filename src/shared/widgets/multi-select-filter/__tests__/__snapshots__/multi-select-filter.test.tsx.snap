// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelectFilter /> renders correctly 1`] = `
<div>
  <div
    class="app__multi-select-filter"
  >
    <h2
      class="app__multi-select-filter__title"
    >
      Multi-Select Filter
    </h2>
    <div
      class="ui__card"
    >
      <p
        class="app__multi-select-filter__subtitle"
      >
        Product group:
         
        fantasy
      </p>
      <form>
        <div
          class="ui__input"
        >
          <i
            class="ui__input__icon"
          >
            <svg
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.8 12.2l-3.9-4c.6-.8 1.1-2 1.1-3.1 0-2.8-2.3-5-5-5-2.8 0-5 2.3-5 5 0 2.8 2.2 5 5 5 .8 0 1.7-.2 2.3-.5l3.9 4.3c.2.2.6.2.8 0l.8-.8c.3-.3.3-.7 0-.9zM6 8.1c-1.7 0-3-1.4-3-3 0-1.7 1.4-3 3-3s3 1.4 3 3-1.4 3-3 3z"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          <input
            class="ui__input__bar"
            placeholder="Search..."
            value=""
          />
        </div>
        <div
          class="app__multi-select-filter__items"
        >
          <div
            class="app__multi-select-filter__items__item"
            data-testid="Fantasy Selected"
          >
            <div
              class="app__multi-select-filter__items__item__checkbox app__multi-select-filter__items__item__checkbox--selected"
            />
            <span
              class="app__multi-select-filter__items__item__text app__multi-select-filter__items__item__text--selected"
            >
              <span>
                Fantasy Selected
              </span>
            </span>
          </div>
          <div
            class="app__multi-select-filter__items__item"
            data-testid="Fantasy"
          >
            <div
              class="app__multi-select-filter__items__item__checkbox"
            />
            <span
              class="app__multi-select-filter__items__item__text"
            >
              <span>
                Fantasy
              </span>
            </span>
          </div>
        </div>
        <div
          class="app__multi-select-filter__buttons"
        >
          <button
            class="ui__button app__multi-select-filter__buttons__apply"
            data-testid="button_apply"
            type="submit"
          >
            Apply
          </button>
          <button
            class="ui__button app__multi-select-filter__buttons__clear"
            data-testid="button_clear"
          >
            <svg
              class="app__multi-select-filter__buttons__trash-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 6v18h18v-18h-18zm5 14c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 1-1s1 .448 1 1v10zm4-18v2h-20v-2h5.711c.9 0 1.631-1.099 1.631-2h5.315c0 .901.73 2 1.631 2h5.712z"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
